src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -274,11 +286,35 @@ public static void sendExtras(Bundle extras) {
         }
     }
 
+	/*
+     * Retrives badge count from SharedPreferences
+     */
+	public static int getApplicationIconBadgeNumber(Context context){
+        SharedPreferences settings = context.getSharedPreferences("badge", Context.MODE_PRIVATE);
+        int badge = settings.getInt("badge", 0);
+        return badge;
```

Comment:
No need to create a temporary value here unless you are going to log the badge number. You can just return the value of `settings.getInt()`.